### PR TITLE
Developer App Analytics page for Hybrid Org #105

### LINKED ISSUE #105: Error on Developer App Analytics page for Hybrid Org
## Description
After creating a App, On Analytics page I am seeing below error:-
"Unable to retrieve analytics data. Please try again."

On checking Drupal logs I saw below error message:-
"Apigee\Edge\Exception\ClientErrorException: INVALID_ARGUMENT: Invalid JSON payload received. Unknown name "_optimized": Cannot bind query parameter. Field 'optimized' could not be found in request message. in Apigee\Edge\HttpClient\Plugin\ResponseHandlerPlugin->decodeResponse() (line 104 of G:\Projects\apigee_kickstart\vendor\apigee\apigee-client-php\src\HttpClient\Plugin\ResponseHandlerPlugin.php)."

Refer screenshot for Apigee Debug message of Request and Response

## Version Info
2.0.5

image
image

### PR DESCRIPTION
**Pull request by phdhiren**
Analytics page for Hybrid org, there was change in API end points for hybrid orgs

Reference issues:

https://github.com/apigee/apigee-client-php/issues/105
https://github.com/apigee/apigee-edge-drupal/issues/552
https://github.com/apigee/apigee-edge-drupal/issues/481

### CODE REVIEW

File: src/Api/Management/Controller/StatsController.php [hunk 1]
```diff
@@ -68,22 +74,35 @@ public function __construct(string $environment, string $organization, ClientInt
         // Return responses as an associative array instead of in Apigee Edge's mixed object-array structure to
         // make developer's life easier.
         $this->jsonDecoder = new JsonDecode(true);
+        $this->optimized_js = false;
+        $this->is_hybrid = false;
```
Comment (mxr576) [inline src/Api/Management/Controller/StatsController.php]: 1. Properties should be camel cased
2.  Why don't you set the value here?
`ClientInterface::HYBRID_ENDPOINT === $this->getClient()->getEndpoint();` Or if that is not possible, then this should not be a property rather a helper method.

File: src/Api/Management/Controller/StatsController.php [hunk 2]
```diff
@@ -68,22 +74,35 @@ public function __construct(string $environment, string $organization, ClientInt
...
-        $query_params = [
+        $this->is_hybrid = ClientInterface::HYBRID_ENDPOINT === $this->getClient()->getEndpoint();
+        $query_params = [];
+        if ('js' === $optimized && !$this->is_hybrid) {
+            $query_params = [
                 '_optimized' => $optimized,
             ] + $this->normalizer->normalize($query);
+        } else {
+            $query_params = $this->normalizer->normalize($query);
+            $this->optimized_js = true;
```
Comment (mxr576) [inline src/Api/Management/Controller/StatsController.php]: I do not see why this should be stored in a property, can you elaborate?

File: src/Api/Management/Controller/StatsController.php (hunk 1 above)
Comment (phdhiren) [reply src/Api/Management/Controller/StatsController.php]: I've moved that to separate helper method

File: src/Api/Management/Controller/StatsController.php (hunk 2 above)
Comment (phdhiren) [reply src/Api/Management/Controller/StatsController.php]: With updated logic property not required

**Review by raakesh-blokhra — APPROVED** [review]
LGTM thanks

**Comment by phdhiren** [comment]
The pull request will fix https://github.com/apigee/apigee-edge-drupal/issues/481

**Pull request merged**